rclpy/rclpy/node.py

```diff
@@ -247,6 +247,8 @@ def destroy_timer(self, timer):
         for tmr in self.timers:
             if tmr.timer_handle == timer.timer_handle:
                 _rclpy.rclpy_destroy_entity(tmr.timer_handle)
+                # TODO(sloretz) Store clocks on node and destroy them separately
+                _rclpy.rclpy_destroy_entity(tmr._clock._clock_handle)
```

Comment:
A ROSTime clock would need to be fed from a time source that is tied to the node. Clocks may need to be gotten off a `Node` instance, so it would make sense for that instance to store and destroy them later.